stackit/internal/services/ske/cluster/resource.go

```diff
@@ -657,7 +658,11 @@ func (r *clusterResource) createOrUpdateCluster(ctx context.Context, diags *diag
 
 func (r *clusterResource) getCredential(ctx context.Context, model *Model) error {
 	c := r.client
-	res, err := c.GetCredentials(ctx, model.ProjectId.ValueString(), model.Name.ValueString()).Execute()
+	res, err := c.CreateKubeconfig(ctx, model.ProjectId.ValueString(), model.Name.ValueString()).
```

Comment:
This can be simplified to:

```
payload := ske.CreateKubeconfigPayload{
		ExpirationSeconds: utils.Ptr(DefaultKubeconfigExpiration),
	}
	res, err := c.CreateKubeconfig(ctx, model.ProjectId.ValueString(), model.Name.ValueString()).CreateKubeconfigPayload(payload).Execute()
```